providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -1,29 +1,296 @@
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue,
+        EvaluationContext ctx) {
+        ResolveStringRequest request = ResolveStringRequest.newBuilder()
+            .setFlagKey(key)
+            .setContext(this.convertContext(ctx)).build();
+        ResolveStringResponse r = this.serviceStub.resolveString(request);
+        return ProviderEvaluation.<String>builder().value(r.getValue())
+            .variant(r.getVariant())
+            .reason(this.mapReason(r.getReason()))
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue,
+        EvaluationContext ctx) {
+        ResolveFloatRequest request = ResolveFloatRequest.newBuilder()
+            .setFlagKey(key)
+            .setContext(this.convertContext(ctx))
+            .build();
+        ResolveFloatResponse r = this.serviceStub.resolveFloat(request);
+        return ProviderEvaluation.<Double>builder()
+            .value(r.getValue())
+            .variant(r.getVariant())
+            .reason(this.mapReason(r.getReason()))
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue,
+        EvaluationContext ctx) {
+        ResolveIntRequest request = ResolveIntRequest.newBuilder()
+            .setFlagKey(key)
+            .setContext(this.convertContext(ctx))
+            .build();
+        ResolveIntResponse r = this.serviceStub.resolveInt(request);
+        return ProviderEvaluation.<Integer>builder()
+            .value((int) r.getValue())
+            .variant(r.getVariant())
+            .reason(this.mapReason(r.getReason()))
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<Structure> getObjectEvaluation(String key, Structure defaultValue,
+        EvaluationContext ctx) {
+        ResolveObjectRequest request = ResolveObjectRequest.newBuilder()
+            .setFlagKey(key)
+            .setContext(this.convertContext(ctx))
+            .build();
+        ResolveObjectResponse r = this.serviceStub.resolveObject(request);
+        return ProviderEvaluation.<Structure>builder()
+            .value(this.convertObjectResponse(r.getValue()))
+            .variant(r.getVariant())
+            .reason(this.mapReason(r.getReason()))
+            .build();
     }
 
-    /** 
-     * A test method.
-     * 
-     * @return {boolean}
+    // Map FlagD reasons to Java SDK reasons.
+    private Reason mapReason(String flagDreason) {
+        if (!EnumUtils.isValidEnum(Reason.class, flagDreason)) {
+            // until we have "STATIC" in the spec and SDK, we map STATIC to DEFAULT
+            if ("STATIC".equals(flagDreason)) {
+                return Reason.DEFAULT;
+            } else {
+                return Reason.UNKNOWN;
+            }
+        } else {
+            return Reason.valueOf(flagDreason);
```

Comment:
I should add a test for this now that you mention it, but this checks that `flagDreason` (a string) can be mapped to a `Reason` enum value, and if it can't it returns `Reason.UNKNOWN`.

There's a special case for now with `"STATIC"`, which we want to eventually add to the spec, and which we're mapping to `Reason.DEFAULT` for now.

Reply:
Added a test for this.